### DISCUSSION TITLE
KRIDA 4CH AC dimmer wiring

### DISCUSSION THREAD

**Discussion by lophital** [theyosh/TerrariumPI#767] [Q&A] [answered]
Hi, I guess I am the first one who posts questions here in 2023.
Happy new year to all :) 
In short, my questions are about pinning and wiring KRIDA 4CH AC dimmer.
I couldn't find any guide or pictures for advice from google. It ended up to asking for a help in here.
They have GND, VCC, SYNC, CH1, CH2.... etc pins but not sure how can I connect them and detect on terrariumPi.
I had no problem with BMP280 sensor for auto-detection but this AC dimemr is hard part.

My second question is what should I learn and what else devices are needed for following set ups? 
So my plan is running 12 enclosures with 24 dimmer(12 for heater, 12 for light), 12 Relay(12 misting nozel).
So I guessed I would need 12 BME280 sensors and purchased 4CH, 8CH dimmers from KRIDA.
I am still not clear with what I am doing and all terms like I2C things. 
It would be really appreciated if someone here could let me know how to start to learn for beginners like me. 
I am really interested and willing to translate this cool project into Korean, but all these hardware things are really new to me.

Thanks in advance.

**Comment by theyosh** [comment]
Connecting the dimmer you need 4 wires. These are needed to connect to the I2C bus of the PI. https://pinout.xyz/pinout/i2c#

So you need to connect the following wires from the dimmer device to the PI:

VCC -> Pin 2 https://pinout.xyz/pinout/5v_power#
GND -> Pin 6 https://pinout.xyz/pinout/ground#
SDA -> Pin 3 https://pinout.xyz/pinout/pin3_gpio2#
SCL -> Pin 5 https://pinout.xyz/pinout/pin5_gpio3#

Then power your PI and see if you can find it at address `3f` or `27` following: https://theyosh.github.io/TerrariumPI/hardware/#i2c-bus If this is the case, the device is connected to the Pi and can be used by the software.

**Than you need to add 220Volt power devices. Pay attention, this is very dangerous, as you are working with live wire!!!!**

On the first blue connector you put your power. So 2 wires and a wall connector goes there. On the other blue connectors, you can hook up your electronic device. They are named CH1-4

Follow this manual on how to add them to TP4 https://theyosh.github.io/TerrariumPI/hardware/relay/i2c-4channels-led-ac-dimmer/

Then, controlling 12 enclosures is a but much. I have seen people controlling 8 enclosures, which I was a bit amazed about. I do not know how much enclosures it can handle.

But you will get problems with your hardware. This is because the BME280 can only use 2 different I2C addresses, which means at max, you can add 2 times a BME280 sensor. Not 12. In order to add 12 BME280 sensors, you need to add 5 new I2C busses on the Raspberry. And this is rather tricky. A way is using multiplexers: https://theyosh.github.io/TerrariumPI/hardware/i2c-multiplexer/ But this will only allow adding 8 BME280 sensors. So, that means you have to create a second I2C bus on the RPI self. And that is also possible, but rather difficult.

So, the software is made to support multiple enclosures, but the RPI hardware could be the limiting factor here. So in your case, I would go for at least 2 Raspberry PIs and control 6 enclosures each.

Also, it depends on the used sensors. Some sensors take some time for a measurement. And if all measurements takes more then 30 seconds, you will run in problems. So 12 enclosures sound a bit much.

And finally, my software does require that you are bit handy with computers and the hardware. In your case, maybe it is better just to start with one enclosure first. Just to see how everything works.

**Comment by lophital** [comment, marked as answer]
Oh god, thanks a lot for the advice! I kept watching why I don't have SDA and SCL parts on dimmer and I just noticed what I have is V2 version for 4ch dimmer and V3 version for 8ch dimmer... That was totally different product I guess. I was dumb.
I just ordered the product in the link("classic" 4ch dimmer), additional RPI, and multiplexers! Really thanks a lot again and I will try these tips once they arrive here! 
Yes, I am just familiar with R language which is quite far away from python and JS. But this project looks really fun and cool. The pilot test is really necessary and I will try a simple one first.  I guess it will work well soon. I will post the photos here https://github.com/theyosh/TerrariumPI/issues/210 with the finished job :)
